src/config/proxy.ts

```diff
@@ -1,15 +1,27 @@
 import { envVars }  from "./env";
-import { bootstrap } from "global-agent";
-import { getLogger } from "./logger";
-
-const logger = getLogger("config.proxy");
-
-if (envVars.PROXY) {
-	logger.info(`configuring proxy: ${envVars.PROXY} for outbound calls`);
-	process.env.GLOBAL_AGENT_HTTP_PROXY = envVars.PROXY;
-	bootstrap();
-} else {
-	logger.info("configuring no proxy for outbound calls");
-}
+import { AxiosRequestConfig } from "axios";
+import { HttpsProxyAgent } from "https-proxy-agent";
+
+const outboundProxyHttpsAgent = envVars.PROXY ? new HttpsProxyAgent(envVars.PROXY) : undefined;
+
+/**
+ * Use this Axios config to make HTTP(S) requests via the configured outbound proxy.
+ */
+export const OUTBOUND_PROXY_CONFIG: Partial<AxiosRequestConfig> = {
```

Comment:
I don't like the idea that this util knows about AxiosRequestConfig data type.  

Plus, this is only being used in a single place right? do we even need this file at all? We could put all of it in `github-client`

Reply:
moved the proxy config into the GitHubClient